src/enums.h

```diff
@@ -363,4 +363,26 @@ std::array<T, N> vec_to_array(const std::vector<T> &vec)
     return array;
 }
 
+template <typename T>
```

Comment:
I believe I recall reading somewhere that `std::begin(container)` is preferable to `container.begin()` (same for `end`), but I couldn't begin to say why.